src/main/scala/org/constellation/primitives/Schema.scala

```diff
@@ -194,7 +195,15 @@ object Schema {
   case class ObservationEdge( // TODO: Consider renaming to ObservationHyperEdge or leave as is?
     parents: Seq[TypedEdgeHash],
     data: TypedEdgeHash
-  ) extends Signable
+  ) extends Signable {
+
+    override def getRunLengthEncoding = {
+      val numParentsLengthString = parents.length.toString
+      val parentValuesString = parents.flatMap(parent => parent.hashReference.length :: parent.hashReference :: Nil).mkString("")
+      val runEncodedTxData = data.hashReference//todo call this something else? Payload?
+        numParentsLengthString :: parentValuesString :: runEncodedTxData :: Nil mkString ""
+    }
```

Comment:
Considering above function, it will look like this:
```
runLengthEncoding(parents.map(_.hashReference) ++ List(data.hashReference): _*)
```

Reply:
Yeah and I'm using Cons anyhow